### DISCUSSION TITLE
Numbers of new posts shown by Tag Tracking

### DISCUSSION THREAD

**Discussion by arcanetrivia** [AprilSylph/XKit-Rewritten#336] [Lounge]
tl;dr I'm curious which version of the extension is placing each of the three new posts counts I see, and how they are calculated.

I use the Tag Tracking+ extensions from both New XKit and XKit Rewritten simultaneously, because I like to have the tracked tags in the sidebar rather than having to click through the search bar. (It would be great if Rewritten could have an option to do this, but I gather you prefer to refrain from inserting elements directly into the page whenever possible?) So this is a fairly typical display for me if I do click in the search bar and look at my tracked tags there:

![tagtracking](https://user-images.githubusercontent.com/68124115/142284574-b9fb932e-f31f-4e30-b04f-0afe2efb0c79.PNG)

I'm not quite sure why I'm seeing three counts for each tag, and what data is being drawn on to calculate each of them. The ones in the sidebar obviously come from New XKit. And I _think_ that is also where the number in parentheses after "takin' over the asylum" and "varda" is coming from, although I have the "Show a [new] indicator in the tag search bar" option turned off (or is that not what it does?). I therefore guess that the number in grey along the right edge of the search dropdown is coming from XKit Rewritten. Is that true?

So, if that is the case - we can see that these numbers don't agree. So I'm curious what the two versions of the extension are looking at to determine if I have any unseen posts in a given tag. In New XKit, it's possible for the indicator to get stuck showing a number when I can't in fact see any new post if I visit the tag (takin' over the asylum has been stuck like this for some months now). My best guess has been that there _is_ a post, but I can't see it because I have the poster blocked (or possibly vice versa). Thus it never gets "marked read" and the indicator stays stuck until a post appears that I can see. (I've occasionally gotten so annoyed with this that I made a throwaway post myself just to clear the indicator.) But the number I assume to be coming from XKit Rewritten for the same tag is 0, so that makes me think it's not calculating the number in the same way.

(But I _can_ see the new post that's in #varda - so I don't know why it was showing 0, because the truth was there was in fact 1 new post...)

I haven't yet encountered the situation of the number seeming to get stuck in Rewritten, so maybe the way it calculates dates would make such a feature unnecessary, but - what about a feature to force-clear the new post count?

**Comment by AprilSylph** [comment]
> I therefore guess that the number in grey along the right edge of the search dropdown is coming from XKit Rewritten. Is that true?

Correct.

> But the number I assume to be coming from XKit Rewritten for the same tag is 0, so that makes me think it's not calculating the number in the same way.

Also correct; New XKit's Tag Tracking+ uses the raw API, while XKit Rewritten checks the same endpoint that the new dash uses to display the posts (thus ensuring its perfect accuracy).

> what about a feature to force-clear the new post count?

The new post count is calculated only when you open the search dropdown, counting 1 for every post it finds up until the last post you saw on that tag page. Dates/times are not saved, only post IDs. You could clear it by modifying your config data manually, but this would result in the count matching however many posts are found on the first "page" of the tag every time.
